rfcs/0000-nixpkgs-breaking-change-policy.md

```diff
@@ -0,0 +1,75 @@
+# nixpkgs Breaking Change Policy
+
+| feature          | start-date | author    | co-authors                                    | shepherd-team                                                   | shepherd-leader                                  | related-issues                             |
+| ---------------- | ---------- | --------- | --------------------------------------------- | --------------------------------------------------------------- | ------------------------------------------------ | ------------------------------------------ |
+| breaking_changes | 2020-09-24 | Kevin Cox | (find a buddy later to help out with the RFC) | (names, to be nominated and accepted by RFC steering committee) | (name to be appointed by RFC steering committee) | (will contain links to implementation PRs) |
+
+# Summary
+
+This defines a policy for how to manage nixpkgs changes that break dependent derivations or tests. It specifies time frames and procedures so that maintainers of dependencies and dependents can know what to expect from each other when making breaking changes.
+
+This document is only focused on changes to derivation builds and tests it **does not** aim to make any opinion on changes that can break user machines or configurations.
+
+# Motivation
+
+Right now the procedure for making breaking changes varies widely by maintainer and is largely tribal knowledge. There are a wide variety of approaches ranging from:
+
+- After testing to confirm that a change is correct, merge to `master` and notify dependents that they need to update their packages.
```

Comment:
The initial target branch that is chosen depends on several factors
- awareness of the staging branch and what its for
- knowing the amount of rebuilds needed and thereby whether it should target staging or master
- whether requires changes are on any of the possible target branches

Reply:
This is largely what this policy is trying to address. We are trying to write down what should be done to make these choices better known and more clear.